drawing_shape2

```diff
@@ -0,0 +1,201 @@
+    - jak ponownie wywołać rysowanie skoro już zostało zakończone?
+    - jak zorganizować całosć kodu?
+"""
+
+from PyQt5.QtWidgets import QWidget, QApplication, QGridLayout, QLabel, QMessageBox # QGridLayout, QLabel - do obsługi myszy
+from PyQt5.QtGui import QPainter, QPen, QPainterPath
+from PyQt5.QtCore import Qt
+import sys
+
+class Mapa(QWidget):
+    # czy będzie dobrym pomysłem zdefiniowanie zmiennej klasowej tutaj (przed funkcjami) zawierającej informację o wymiarach okna (w i h)?
+    def __init__(self):
+        super().__init__()
+        self.h = 800    #zdefiniowanie h i w jako zmiennych klasowych (wysokosc i szerokosc okna)
+        self.w = 900
+        self.initUI()
+        
+    def initUI(self):     
+
+        self.setGeometry(150, 150, self.w, self.h-120) #położenie okna poziom, pion, rozmiary okna poziom, pion
+                                                        #rozmiar okna w pionie zmiejszony o 130 - wspołrzędną NS Opołonka w PUWG1992, aby pod mapą miejsca nie zostawiać
+        self.setWindowTitle('Walory v.1.0 by KP')
+        self.show()
+
+        
+        grid = QGridLayout()#ustawienie siatki do wyswietlania elementow na mapie
+        grid.setSpacing(10)
+        
+        x = 0
+        y = 0
+        
+        self.text = "x: {0},  y: {1}".format(x, y)
+        
+        self.label = QLabel(self.text, self)
+        grid.addWidget(self.label, 0, 0, Qt.AlignTop)
+        
+        #self.setMouseTracking(False) #jesli true, to sledzi mysz nawet bez kliknięcia; jesli False to sledzi ruch z wcisniętym klawiszem (ale nie reaguje na kliknięcie) 
+        self.setLayout(grid)
+                
+               
+    def paintEvent(self, e):    #funcja wywoływana automatycznie, odpowiada za całosć rysowania
+        
+        self.drawLines()  #wywołanie rysowania linii
+        self.drawPoints() #wywołanie rysowania miejscowosci
+        
+        
+    def drawLines(self):
+        """
+        pomysł na dopasowywanie rozmiaru obrazu do rozmiaru okna:
+        size = self.size() #funkcja pobierająca rozmiar okna - do wykorzystania w innym miejscu
+        w = size.width()
+        h = size.height() # wartosc wysokosci okna (musi byc zgodna z podana w initUI)
+       
+        """
+        h = self.h # przypisanie wartosci zmiennej klasowej self.h do zmiennej funkcji h 
+       
+        ### rysowanie wojewodztw
+        #pętla rysująca kształty z shp - poligony, za pętlą instrukcja połączenia ostatniego punktu z pierwszym
+        
+        qp = QPainter() #przypisanie QPaintera do qp
+        qp.begin(self)  #rozpoczęcie QPaintera
+        
+        path = QPainterPath()   #rozpoczęcie sciezki
+        pen = QPen(Qt.black, 1, Qt.SolidLine) #zdefiniowanie stylu piora
+        qp.setPen(pen)          #ustanowienie piora ze stylem pen
+        
+        for j in range (len(shap_woj)): #pętla rysująca j kształtow ze zmiennej shap_woj (wczytanej niżej przez shapefile.Reader)
```

Comment:
Pierwsza rzecz - logika rysująca województwa i rysująca rzeki wygląda tak samo. Dałoby się wydzielić.
Druga - odwołujesz się do zmiennej, której tutaj nie ma. Jeśli chcesz ją załadować w innej funkcji, to zrób to na polu klasy - czyli powinno być `self.shap_woj`